include/seqan3/range/view/take.hpp

```diff
@@ -117,31 +117,31 @@ class view_take : public ranges::view_interface<view_take<urng_t, exactly, or_th
          * \brief seqan3::detail::inherited_iterator_base operators are used unless specialised here.
          * \{
          */
-        iterator_type & operator++() noexcept(noexcept(++base_base_t{}))
+        iterator_type & operator++() noexcept(noexcept(++std::declval<base_base_t &>()))
         {
-            base_base_t::operator++();
+            base_t::operator++();
             ++pos;
             return *this;
         }
 
-        iterator_type operator++(int) noexcept(noexcept(++base_base_t{}))
+        iterator_type operator++(int) noexcept(noexcept(++std::declval<base_base_t &>()))
```

Comment:
```suggestion
        iterator_type operator++(int) noexcept(noexcept(++std::declval<base_base_t &>()++))
```

Reply:
I think this was not resolved and need some similar strategy then the changes you made for the inherited_iterator_base class. So actually test the operations of the function and not what the base type would do. BTW I noticed that the constructor definitions are all marked noexcept explicitly. If I am not mistaken this depends on whether or not the base iterator is noexcept constructible since it is inheriting from it.

Reply:
No, this only depends on the copy constructability of the iterator itself. I see no reason why the iterator should not be noexcept copy cosntructable, but I have added a check to make you happy.

Reply:
Because it implies that also all it's base classes need to be noexcept copy constructible. And while you enforce it for the derived class it is not enforced by the base class. And even worse, the base class inherits from another base class which is a template and we have no control over it. So copying the derived class will naturally call copy constructor of the base class which can possibly throw. So in the rare case that there is a potential throwing base iterator, we will get compiler errors with gcc9 wouldn't we? Or is there something that prevents this?

Reply:
Inheriting constructors have an exception specification according to the stuff they are inheriting. If the base copy constructor is not noexcept, than the derived copy constructor won't be either:
https://stackoverflow.com/questions/18653726/is-the-default-move-constructor-defined-as-noexcept

Reply:
I think we are on totally different pages.
This is what I am referring to: https://godbolt.org/z/Ha-AEo
The iterator class in question has all it's constructors declared noexcept explicitly, while it's derived class (inherit_iterator_base) has not! This means, that we get a hard compiler error with gcc9 when the base_t iterator throws on construction (the one which comes from outside and we don't know about). So the explicit noexcept specification of this iterator class is wrong and should be just removed. This also means that there is no guarantee that post in/decrement if doing a copy of `this` is noexcept, unless we explicitly test for it. 

But anyway I will look into the code as soon as possible (like in the late evning).

Reply:
You are absolutely right, I guess I was in the office for too long on Friday :wink: